common/djangoapps/student/tests/test_login.py

```diff
@@ -78,10 +81,18 @@ def test_login_success_unicode_email(self):
         self.user.email = unicode_email
         self.user.save()
 
-        response, mock_audit_log = self._login_response(unicode_email, 'test_password', patched_audit_log='student.models.AUDIT_LOG')
+        response, mock_audit_log = self._login_response(unicode_email, 'test_password',
+                                                        patched_audit_log='student.models.AUDIT_LOG')
         self._assert_response(response, success=True)
         self._assert_audit_log(mock_audit_log, 'info', [u'Login success', unicode_email])
 
+    def test_login_success_prevent_auth_user_writes(self):
+        with waffle().override(PREVENT_AUTH_USER_WRITES, True):
+            old_last_login = self.user.last_login
+            self.test_login_success()
+            self.user.refresh_from_db()
+            assert old_last_login == self.user.last_login
```

Comment:
Is there a test that checks if last_login *is* updated with a False waffle flag? I couldn't find one.

Reply:
No, I guess it makes sense to add that since we're replacing the signal handler.  I did test that manually to make sure it worked, but I'll add an explicit test for that.